spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -1030,16 +1249,22 @@
 
   describe "#destroy_vm_and_pg" do
     it "deletes resources and exits" do
-      expect(postgres_server.vm).to receive(:incr_destroy)
-      expect(postgres_server).to receive(:destroy)
+      vm_strand = postgres_server.vm.strand
+      pg_server_id = postgres_server.id
+      expect(Semaphore.where(strand_id: vm_strand.id, name: "destroy").count).to eq(0)
 
       expect { nx.destroy_vm_and_pg }.to exit({"msg" => "postgres server is deleted"})
+
+      # Verify vm.incr_destroy was called
+      expect(Semaphore.where(strand_id: vm_strand.id, name: "destroy").count).to eq(1)
+      # Verify postgres_server was destroyed
+      expect(PostgresServer[pg_server_id]).to be_nil
```

Comment:
I think this is the simpler rspec syntax:

```suggestion
      expect(postgres_server).not_to exist
```